kv/kvserver: TestClosedTimestampInactiveAfterSubsumption failed

[(kv/kvserver).TestClosedTimestampInactiveAfterSubsumption failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2331021&tab=buildLog) on [release-20.2@22d91d4e924cabd6e45443c76110cbcef9a43e93](https://github.com/cockroachdb/cockroach/commits/22d91d4e924cabd6e45443c76110cbcef9a43e93):

```
I200929 18:39:43.745019 747792 kv/kvserver/replica_command.go:1654  [n1,replicate,s1,r1/1:/{Min-System/NodeL…}] change replicas (add [(n2,s2):3] remove []): existing descriptor r1:/{Min-System/NodeLiveness} [(n1,s1):1, (n3,s3):2, (n2,s2):3LEARNER, next=4, gen=3]
I200929 18:39:43.774174 747792 kv/kvserver/replica_raft.go:247  [n1,s1,r1/1:/{Min-System/NodeL…}] proposing SIMPLE(v3) ADD_REPLICA[(n2,s2):3]: after=[(n1,s1):1 (n3,s3):2 (n2,s2):3] next=4
I200929 18:39:43.827118 747792 testutils/testcluster/testcluster.go:805  WaitForFullReplication took: 3.232330617s
I200929 18:39:43.861011 752527 sql/event_log.go:146  [n1,client=127.0.0.1:55458,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.closed_timestamp.target_duration Value:00:00:05 User:root}
I200929 18:39:43.878240 752527 sql/event_log.go:146  [n1,client=127.0.0.1:55458,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.closed_timestamp.close_fraction Value:0.333 User:root}
I200929 18:39:43.903115 752527 sql/event_log.go:146  [n1,client=127.0.0.1:55458,hostssl,user=root] Event: "set_cluster_setting", target: 0, info: {SettingName:kv.closed_timestamp.follower_reads_enabled Value:true User:root}
I200929 18:39:43.916638 752527 sql/event_log.go:146  [n1,client=127.0.0.1:55458,hostssl,user=root] Event: "create_database", target: 52, info: {DatabaseName:cttest Statement:CREATE DATABASE cttest User:root}
I200929 18:39:43.958665 752527 sql/event_log.go:146  [n1,client=127.0.0.1:55458,hostssl,user=root] Event: "create_table", target: 53, info: {TableName:cttest.public.kv Statement:CREATE TABLE cttest.public.kv (id INT8 PRIMARY KEY, value STRING) User:root}
I200929 18:39:43.961080 752495 kv/kvserver/replica_command.go:414  [n1,split,s1,r35/1:/{Table/39-Max}] initiating a split of this range at key /Table/53 [r36] (zone config)
I200929 18:39:47.132007 747792 kv/kvserver/closed_timestamp_test.go:908  SucceedsSoon: [NotLeaseHolderError] r36: replica (n3,s3):2 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1601404783.999578041,0 epo=1 pro=1601404783.999581832,0
I200929 18:39:48.132686 747792 kv/kvserver/closed_timestamp_test.go:908  SucceedsSoon: [NotLeaseHolderError] r36: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1601404783.999578041,0 epo=1 pro=1601404783.999581832,0
I200929 18:39:49.133366 747792 kv/kvserver/closed_timestamp_test.go:908  SucceedsSoon: [NotLeaseHolderError] r36: replica (n3,s3):2 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1601404783.999578041,0 epo=1 pro=1601404783.999581832,0
I200929 18:39:50.002944 748460 server/status/runtime.go:522  [n1] runtime stats: 522 MiB RSS, 1066 goroutines, 123 MiB/212 MiB/250 MiB GO alloc/idle/total, 25 MiB/84 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (800x), 11 MiB/11 MiB (r/w)net
I200929 18:39:50.134122 747792 kv/kvserver/closed_timestamp_test.go:908  SucceedsSoon: [NotLeaseHolderError] r36: replica (n3,s3):2 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1601404783.999578041,0 epo=1 pro=1601404783.999581832,0
I200929 18:39:50.383249 749567 server/status/runtime.go:522  [n2] runtime stats: 522 MiB RSS, 1070 goroutines, 138 MiB/199 MiB/250 MiB GO alloc/idle/total, 26 MiB/84 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (800x), 14 MiB/14 MiB (r/w)net
I200929 18:39:50.592048 750034 server/status/runtime.go:522  [n3] runtime stats: 523 MiB RSS, 1065 goroutines, 153 MiB/186 MiB/253 MiB GO alloc/idle/total, 27 MiB/85 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (800x), 16 MiB/16 MiB (r/w)net
I200929 18:39:51.134804 747792 kv/kvserver/closed_timestamp_test.go:908  SucceedsSoon: [NotLeaseHolderError] r36: replica (n3,s3):2 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1601404783.999578041,0 epo=1 pro=1601404783.999581832,0
I200929 18:39:52.147749 754436 kv/kvserver/replica_command.go:414  [n2,s2,r36/3:/{Table/53-Max}] initiating a split of this range at key /Table/53/1/2 [r46] (manual)
I200929 18:39:52.193187 747792 testutils/testcluster/testcluster.go:801  WaitForFullReplication
I200929 18:39:52.207102 747792 testutils/testcluster/testcluster.go:805  WaitForFullReplication took: 13.867043ms
I200929 18:39:52.211538 747792 testutils/testcluster/testcluster.go:801  WaitForFullReplication
I200929 18:39:52.227370 747792 testutils/testcluster/testcluster.go:805  WaitForFullReplication took: 15.786929ms
I200929 18:39:52.227993 747792 util/stop/stopper.go:563  quiescing
I200929 18:39:52.228051 754533 util/stop/stopper.go:563  quiescing
I200929 18:39:52.228306 754531 util/stop/stopper.go:563  quiescing
W200929 18:39:52.228354 748383 kv/kvserver/store.go:1896  [n1,s1,r36/1:/Table/53{-/1/2}] unable to gossip on capacity change: node unavailable; try another peer
I200929 18:39:52.228472 754532 util/stop/stopper.go:563  quiescing
W200929 18:39:52.228757 750047 sql/sqlliveness/slinstance/slinstance.go:182  [n3] exiting heartbeat loop
W200929 18:39:52.228888 750141 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200929 18:39:52.229178 749837 gossip/gossip.go:1494  [n3] no incoming or outgoing connections
W200929 18:39:52.229362 749608 sql/sqlliveness/slinstance/slinstance.go:182  [n2] exiting heartbeat loop
W200929 18:39:52.229489 750227 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200929 18:39:52.229524 752556 kv/kvserver/intentresolver/intent_resolver.go:770  failed to gc transaction record: could not GC completed transaction anchored at /Local/Range/Table/53/RangeDescriptor: node unavailable; try another peer
W200929 18:39:52.229576 753830 kv/kvserver/raft_transport.go:637  [n3] while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200929 18:39:52.229695 748505 sql/sqlliveness/slinstance/slinstance.go:182  [n1] exiting heartbeat loop
W200929 18:39:52.236629 749379 gossip/gossip.go:1494  [n2] no incoming or outgoing connections
W200929 18:39:52.236608 750364 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200929 18:39:52.236629 750297 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200929 18:39:52.236770 748612 jobs/registry.go:669  canceling all adopted jobs due to stopper quiescing
W200929 18:39:52.236807 749619 jobs/registry.go:669  canceling all adopted jobs due to stopper quiescing
W200929 18:39:52.236962 753782 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200929 18:39:52.237123 750220 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: EOF:
W200929 18:39:52.237105 750339 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 2: EOF:
W200929 18:39:52.237187 750074 jobs/registry.go:669  canceling all adopted jobs due to stopper quiescing
W200929 18:39:52.237535 750306 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: EOF:
W200929 18:39:52.237611 753825 kv/kvserver/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 3: EOF:
W200929 18:39:52.237748 750211 kv/kvserver/raft_transport.go:637  [n1] while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
    --- FAIL: TestClosedTimestampInactiveAfterSubsumption/with_intervening_lease_transfer (12.39s)
        closed_timestamp_test.go:610: [NotLeaseHolderError] r36: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1601404783.999578041,0 epo=1 pro=1601404783.999581832,0
        closed_timestamp_test.go:601: [NotLeaseHolderError] r36: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):3 seq=2 start=1601404783.999578041,0 epo=1 pro=1601404783.999581832,0

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestClosedTimestampInactiveAfterSubsumption PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestClosedTimestampInactiveAfterSubsumption.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/closed_timestamp_test.go
+++ b/pkg/kv/kvserver/closed_timestamp_test.go
@@ -499,10 +499,7 @@ func TestClosedTimestampInactiveAfterSubsumption(t *testing.T) {
 			callback: nil,
 		},
 		{
-			name: "with intervening lease transfer",
-			// TODO(aayush): Maybe allowlist `TransferLease` requests while a range is
-			// subsumed and use that here, instead of forcing a lease transfer by
-			// pausing heartbeats.
+			name:     "with intervening lease transfer",
 			callback: forceLeaseTransferOnSubsumedRange,
 		},
 	}
@@ -559,38 +556,15 @@ func TestClosedTimestampInactiveAfterSubsumption(t *testing.T) {
 				},
 			},
 		}
-		// If the initial phase of the merge txn takes longer than the closed
-		// timestamp target duration, its initial CPuts can have their write
-		// timestamps bumped due to an intervening closed timestamp update. This
-		// causes the entire merge txn to retry. So we use a long closed timestamp
-		// duration at the beginning of the test until we have the merge txn
-		// suspended at its commit trigger, and then change it back down to
-		// `testingTargetDuration`.
-		tc, db, leftDesc, rightDesc := initClusterWithSplitRanges(ctx, t, 5*time.Second,
+		tc, _, leftDesc, rightDesc := initClusterWithSplitRanges(ctx, t, testingTargetDuration,
 			testingCloseFraction, clusterArgs)
 		defer tc.Stopper().Stop(ctx)
 
 		leftLeaseholder := getCurrentLeaseholder(t, tc, leftDesc)
 		rightLeaseholder := getCurrentLeaseholder(t, tc, rightDesc)
-		if leftLeaseholder.StoreID == rightLeaseholder.StoreID {
-			// In this test, we may pause the heartbeats of the store that holds the
-			// lease for the right hand side range, in order to force a lease
-			// transfer. If the LHS and RHS share a leaseholder, this may cause a
-			// lease transfer for the left hand range as well. This can cause a merge
-			// txn retry and we'd like to avoid that, so we ensure that LHS and RHS
-			// have different leaseholders before beginning the test.
-			target := pickRandomTarget(tc, rightLeaseholder, leftDesc)
-			if err := tc.TransferRangeLease(leftDesc, target); err != nil {
-				t.Fatal(err)
-			}
-			leftLeaseholder = target
-		}
-		// Make sure that the new leaseholder for the left hand side range learns
-		// that it is indeed the leaseholder.
-		require.NoError(t, tc.(*testcluster.TestCluster).WaitForFullReplication())
 
 		g, ctx := errgroup.WithContext(ctx)
-		// Merge the ranges back together. The LHS rightLeaseholder should block right
+		// Merge the ranges back together. The LHS leaseholder should block right
 		// before the merge trigger request is sent.
 		leftLeaseholderStore := getTargetStoreOrFatal(t, tc, leftLeaseholder)
 		blocker := st.BlockNextMerge()
@@ -616,12 +590,6 @@ func TestClosedTimestampInactiveAfterSubsumption(t *testing.T) {
 			t.Fatal(err)
 		case <-time.After(45 * time.Second):
 			t.Fatal("did not receive merge commit trigger as expected")
-		}
-		// Reduce the closed timestamp target duration in order to make the rest of
-		// the test faster.
-		if _, err := db.Exec(fmt.Sprintf(`SET CLUSTER SETTING kv.closed_timestamp.target_duration = '%s';`,
-			testingTargetDuration)); err != nil {
-			t.Fatal(err)
 		}
 		// inactiveClosedTSBoundary indicates the low water mark for closed
 		// timestamp updates beyond which we expect none of the followers to be able

